rotkehlchen/accounting/history_base_entries.py

```diff
@@ -0,0 +1,257 @@
+        taxable=True,
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=True,
+        method='spend',
+        accounting_treatment=TxAccountingTreatment.SWAP,
+    )
+    airdrop_key = str(HistoryEventType.RECEIVE) + '__' + str(HistoryEventSubType.AIRDROP)
+    result[airdrop_key] = HistoryBaseEntrySettings(
+        taxable=LedgerActionType.AIRDROP in pot.settings.taxable_ledger_actions,
+        # count_entire_amount_spend and count_cost_basis_pnl don't matter for acquisitions.
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='acquisition',
+    )
+    reward_key = str(HistoryEventType.RECEIVE) + '__' + str(HistoryEventSubType.REWARD)
+    result[reward_key] = HistoryBaseEntrySettings(
+        taxable=True,
+        # count_entire_amount_spend and count_cost_basis_pnl don't matter for acquisitions.
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='acquisition',
+    )
+    return result
+
+
+def history_base_entries_iterator(
+        events_iterator: Iterator[AccountingEventMixin],
+        associated_event: HistoryBaseEntry,
+) -> Iterator[HistoryBaseEntry]:
+    """
+    Takes an iterator of accounting events and transforms it into a history base entries iterator.
+    Takes associated event as an argument to be able to log it in case of errors.
+    """
+    for event in events_iterator:
+        if not isinstance(event, HistoryBaseEntry):
+            log.error(
+                f'At accounting for event {associated_event.notes} with identifier '
+                f'{associated_event.event_identifier} we expected to take an additional '
+                f'event but found a non history base entry event',
+            )
+            return
+
+        yield event
+
+
+class HistoryBaseEntriesAccountant():
+
+    def __init__(
+            self,
+            evm_accounting_aggregators: 'EVMAccountingAggregators',
+            pot: 'AccountingPot',
+    ) -> None:
+        self.evm_accounting_aggregators = evm_accounting_aggregators
+        self.pot = pot
+        self.tx_event_settings: dict[str, HistoryBaseEntrySettings] = {}
+
+    def reset(self) -> None:
+        self.evm_accounting_aggregators.reset()
+        self.tx_event_settings = (  # Using | operator is fine since keys are unique
+            self.evm_accounting_aggregators.get_accounting_settings(self.pot) |
+            make_default_accounting_settings(self.pot)
+        )
+
+    def process(
+            self,
+            event: HistoryBaseEntry,
+            events_iterator: Iterator[AccountingEventMixin],
+    ) -> int:
+        """Process a history base entry and return number of actions consumed from the iterator"""
+        timestamp = event.get_timestamp_in_sec()
+        type_identifier = event.get_type_identifier()
+        event_settings = self.tx_event_settings.get(type_identifier, None)
+        if event_settings is None:
```

Comment:
shouldn't this check happen only if is an evm event?

Reply:
Yes, thanks